components/governance/org.wso2.carbon.governance.lcm/src/main/java/org/wso2/carbon/governance/lcm/tasks/LCNotificationScheduler.java

```diff
@@ -0,0 +1,227 @@
+                    }
+                }
+            } else {
+                if (log.isDebugEnabled()) {
+                    log.debug("Invalid arguments supplied as lifecycle name: " + lifecycleName + ", lifecycle state: "
+                            + lifecycleState);
+                }
+            }
+        }
+    }
+
+    /**
+     * This method is used get duration differences using two durations in format dd:hh:mm:ss.
+     *
+     * @param timeStampOne  lower duration.
+     * @param timeStampTwo  higher duration.
+     * @return long         duration difference in timestamp format.
+     */
+    private long getDurationDifference(String timeStampOne, String timeStampTwo) {
+        return LifecycleStateDurationUtils.getMillySecondsByDuration(timeStampTwo) - LifecycleStateDurationUtils
+                .getMillySecondsByDuration(timeStampOne);
+    }
+
+    /**
+     * This method is used to calculate notification date of the scheduler.
+     *
+     * @param durationDifference    duration difference of the lifecycle checkpoint.
+     * @return                      notification date in the format of yyyy-MM-dd.
+     */
+    private String getNotificationDate(long durationDifference) {
+        // Getting current time
+        long currentTimeInMillySeconds = Calendar.getInstance().getTimeInMillis();
+
+        Calendar calendar = Calendar.getInstance();
+        calendar.setTimeInMillis(durationDifference + currentTimeInMillySeconds - durationOfADay);
+        int mYear = calendar.get(Calendar.YEAR);
+        int mMonth = calendar.get(Calendar.MONTH);
+        int mDay = calendar.get(Calendar.DAY_OF_MONTH);
+
+        StringBuilder dateBuilder = new StringBuilder(dateLength);
+        dateBuilder.append(mYear);
+        dateBuilder.append(dateSeparator);
+        // +1 is because mMonth is the completed months.
+        dateBuilder.append(mMonth + 1);
+        dateBuilder.append(dateSeparator);
+        dateBuilder.append(mDay);
+        return dateBuilder.toString();
+    }
+
+    /**
+     * This method is used to create notification message.
+     *
+     * @param schedulerBean     scheduler bean need to create the notification message.
+     * @return                  notification message.
+     */
+    private String getNotificationMessage(LCNotification schedulerBean) {
+
+        String[] pathParams = schedulerBean.getRegPath().split("/");
+        // Lifecycle cannot be applied to registry path "/". Hence pathParams.length is always greater than or equals
+        // to one. Hence (pathParams.length - 1) cannot be a minus value.
+        String resourceName = pathParams[pathParams.length - 1];
+
+        StringBuilder stringBuilder = new StringBuilder("Resource '").append(resourceName).append
+                ("'s lifecycle '").append(schedulerBean.getLcName()).append("' is reaching ").append("lifecycle "
+                + "checkpoint '").append(schedulerBean.getLcCheckpointId()).append("' on ")
```

Comment:
Instead of StringBuilder.append(expr1 + expr2), its a good practice to use
use StringBuilder.append(expr1).append(expr2)
explanation available at [1].
This code creates another StringBuilder to do job of "+".

[1] http://blog.eyallupu.com/2010/09/under-hood-of-java-strings.html